### PR TITLE
mgr/orch: logging - handle lists output

### PR DESCRIPTION
**Pull request by shyukri**
Fixes: https://tracker.ceph.com/issues/42769

Signed-off-by: Shyukri Shyukriev <shshyukriev@suse.com>

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/orchestrator.py [hunk 1]
```diff
@@ -559,6 +559,8 @@ def result_str(self):
         """Force a string."""
         if self.result is None:
             return ''
+        if type(self.result) is list:
+            return ' '.join(str(x) for x in self.result)
```
Comment (liewegas) [inline src/pybind/mgr/orchestrator.py]: what about ``\n``?

Comment (shyukri) [reply src/pybind/mgr/orchestrator.py]: Sure, it will increase readability.

File: src/pybind/mgr/orchestrator.py [hunk 2]
```diff
@@ -559,6 +559,8 @@ def result_str(self):
         """Force a string."""
         if self.result is None:
             return ''
+        if type(self.result) is list:
```
Comment (tchaikov) [inline src/pybind/mgr/orchestrator.py]: why not use
```py
if isinstance(self.result, list):
    pretty_print(self.result)
```

**Comment by liewegas** [comment]
This seems to cause a bunch of dashboard and orch test failures: http://pulpito.ceph.com/sage-2020-01-30_22:27:29-rados-wip-sage-testing-2020-01-30-1230-distro-basic-smithi/#

**Comment by shyukri** [comment]
Some of the failures has:
```
2020-01-30T23:42:16.682 INFO:tasks.cephfs_test_runner:FAIL: test_create_osds (tasks.mgr.dashboard.test_orchestrator.OrchestratorControllerTest)
2020-01-30T23:42:16.682 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2020-01-30T23:42:16.682 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2020-01-30T23:42:16.682 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-sage-testing-2020-01-30-1230/qa/tasks/mgr/dashboard/test_orchestrator.py", line 186, in test_create_osds
2020-01-30T23:42:16.682 INFO:tasks.cephfs_test_runner:    self.assertStatus(201)
2020-01-30T23:42:16.682 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-sage-testing-2020-01-30-1230/qa/tasks/mgr/dashboard/helper.py", line 364, in assertStatus
2020-01-30T23:42:16.683 INFO:tasks.cephfs_test_runner:    self.assertEqual(self._resp.status_code, status)
2020-01-30T23:42:16.683 INFO:tasks.cephfs_test_runner:AssertionError: 500 != 201
``` so not sure if it's directly related.

**Comment by tchaikov** [comment]
@shyukri @liewegas i don't think this PR is related to the failure. i think #33014 should address it.

**Pull request merged**